.cbmc-batch/source/utils.c

```diff
@@ -55,12 +55,30 @@ void assert_array_list_equivalence(
     const struct aws_array_list *const lhs,
     const struct aws_array_list *const rhs,
     const struct store_byte_from_buffer *const rhs_byte) {
-    assert(lhs->alloc == rhs->alloc);
-    assert(lhs->current_size == rhs->current_size);
-    assert(lhs->length == rhs->length);
-    assert(lhs->item_size == rhs->item_size);
-    if (lhs->current_size > 0) {
-        assert_byte_from_buffer_matches((uint8_t *)lhs->data, rhs_byte);
+    assert(!lhs == !rhs);
+    if (lhs && rhs) {
+        assert(lhs->alloc == rhs->alloc);
+        assert(lhs->current_size == rhs->current_size);
+        assert(lhs->length == rhs->length);
+        assert(lhs->item_size == rhs->item_size);
+        if (lhs->current_size > 0) {
+            assert_byte_from_buffer_matches((uint8_t *)lhs->data, rhs_byte);
+        }
+    }
+}
+
+void assert_byte_buf_equivalence(
+    const struct aws_byte_buf *const lhs,
+    const struct aws_byte_buf *const rhs,
+    const struct store_byte_from_buffer *const rhs_byte) {
+    assert(!lhs == !rhs);
+    if (lhs && rhs) {
```

Comment:
If we have ```lhs``` but not ```rhs```, what are the sematics.

Reply:
What do you mean? The assertion on the previous line ensures either both are null or both are non-null. Could you clarify the comment?

Reply:
```
if (lhs == rhs) return;
assert (lhs && rhs) // if only one is null, they're different
//they're both non-null, now, check the internals.
```